Formula/rex.rb

```diff
@@ -17,7 +17,7 @@ class Rex < Formula
 
   uses_from_macos "perl"
 
-  if MacOS.version < :big_sur
+  on_catalina :or_older do
```

Comment:
I doubt Homebrew perl has these pre-installed. Could also make sure everything is fine by building with `CI-no-bottles` as only a single formula is modified in this PR.
```suggestion
  on_system :linux, macos: :catalina_or_older do
```

Reply:
Homebrew `perl` has `ExtUtils::MakeMaker` but none of the rest.

I suggest doing
```ruby
uses_from_macos "perl", since: :big_sur
``` 

so we don't have to resources in multiple `on_*` blocks.